final/src/main/java/ua/epam/spring/hometask/strategy/EveryNthStrategy.java

```diff
@@ -0,0 +1,44 @@
+package ua.epam.spring.hometask.strategy;
+
+import ua.epam.spring.hometask.domain.Event;
+import ua.epam.spring.hometask.domain.User;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.LocalDateTime;
+
+public class EveryNthStrategy implements DiscountStrategy {
+
+    private int nthTicketNumber;
+    private double discount;
+
+    @Override
+    public double calculateDiscount(@Nullable User user, @Nonnull Event event,
+                                    @Nonnull LocalDateTime airDateTime, long numberOfTickets) {
+        long bonusTickets = 0;
+
+        for (long i = 1; i <= numberOfTickets; i++) {
+            if (i % nthTicketNumber == 0) {
+                bonusTickets++;
+            }
+        }
```

Comment:
It's not clear what this logic does, rename the variables and move it to separate method.